### PR TITLE
Improve Dockerfile rebuild UX

### PR DESCRIPTION
**Pull request by austinpray**
Hey there! Did a bit of golfing on the dockerfile.

- Reduce the number of layers and clean up after every layer so the docker image is a bit smaller
- Make installing youtube-dl it's own layer because it needs to be more frequently updated than node
- Make installing node_modules it's own layer so npm install is only run when package.json changes

So [try my branch](https://help.github.com/articles/checking-out-pull-requests-locally/) 

```
git fetch git@github.com:austinpray/ytdl-webserver.git dockerfile:austinpray-docker
git checkout austinpray-docker
docker build -t ytdl-webserver-after .
```

Will yield the following image:

```
ytdl-webserver-after                     latest                                     0a06dc23af80        57 seconds ago      746MB
ytdl-webserver-before                    latest                                     ec27e1e72532        31 minutes ago      787MB
```

So actually only ~40MB saved on the image but the main point is that the UX of rebuilding the dockerfile has been improved.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -3,20 +3,27 @@
...
-RUN mkdir -p public/temp
 
-COPY . /home/app
-RUN npm install
-RUN npm run build
+RUN apt-get update \
+    && apt-get install -y curl ffmpeg \
+    && curl -sL https://deb.nodesource.com/setup_6.x | bash - \
+    && apt-get install -y nodejs \
+    && npm install -g npm@latest \
```
Comment (Algram) [inline Dockerfile]: Do we need to manually install npm? Afaik this is included in the node installation we do in `Line 10`

**Comment by Algram** [comment]
Awesome work! :fire: 
I just had one small question, after that I'll merge!

File: Dockerfile (hunk 1 above)
Comment (austinpray) [reply Dockerfile]: You are installing node 6.x which has a way old version of npm

**Comment by austinpray** [comment]
🍻

**Comment by mauritsvdvijgh** [comment]
I was also playing around with the Dockerfile a little bit and noticed that youtube-dl will be installed by the youtube-dl npm package. Might be better to skip the youtube-dl install from apt step.

If you want to update it youtube-dl before installing node packages then install it through pip, it will be more up to date that from the ubuntu repositories.

**Comment by austinpray** [comment]
Yeah I realized that as well and fixed that in my original PR https://github.com/austinpray/ytdl-webserver/blob/c8009177a51c44303aef93ed22eec31ccf6869d7/Dockerfile

But then you went and reduced the image size 20x with your PR so yay!

**Pull request merged**